### PR TITLE
Task 3, Task4, Task5  ADD

### PR DESCRIPTION
**Pull request by ketan-rathod-713**
by mistake created task5 branch from task4 and same with task4 from task3.

### CODE REVIEW

File: Task5/database/connection.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package database
+
+import (
+	"database/sql"
+
+	_ "github.com/lib/pq"
+)
+
+const URL = "postgres://bacancy:admin@localhost/bacancy?sslmode=disable"
```
Comment (Jaimin-Parmar) [inline Task5/database/connection.go]: @ketan-rathod-713 

Make this string to dynamic and use env variables.

Use Enviroment variable for the database creds, database details.

File: Task5/database/connection.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	// defer db.Close()
+
+	CheckError(db.Ping())
+
+	return db
+}
+
+func CheckError(err error) {
+	if err != nil {
+		panic(err)
```
Comment (Jaimin-Parmar) [inline Task5/database/connection.go]: don't use panic in code.

File: Task5/server.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+		fmt.Println("Post Request Successfull")
+
+		firstname := r.FormValue("firstname")
+		lastname := r.FormValue("lastname")
+		dateofbirth := r.FormValue("dateofbirth")
+		email := r.FormValue("email")
+		phone := r.FormValue("phone")
+
+		fmt.Println(firstname, lastname, dateofbirth, email, phone)
```
Comment (Jaimin-Parmar) [inline Task5/server.go]: fmt.Println  use instead log formate.

**Review by Jaimin-Parmar — CHANGES_REQUESTED** [review]

File: Task5/database/connection.go (hunk 1 above)
Comment (ketan-rathod-713) [reply Task5/database/connection.go]: Used godotenv package to load environment variables from .env file in given directory.

File: Task5/server.go (hunk 3 above)
Comment (ketan-rathod-713) [reply Task5/server.go]: Noted ! In recent commit used log format.

File: Task5/server.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	fs := http.FileServer(http.Dir("static"))
+	http.Handle("/", fs) //TODO: StripPrefix creates a new handler. And how it will see it relative to our folder and price.
+
+	// Form Handler
+	http.HandleFunc("/form", formHandler)
+
+	// Users Data Handler
+	http.HandleFunc("/users", usersDataHandler)
+
+	log.Fatal(http.ListenAndServe(":8080", nil)) // Here nil  because we are setting up http2 here hence no need to define it.
```
Comment (Jaimin-Parmar) [inline Task5/server.go]: take this port from env file.

File: Task5/database/connection.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	if err != nil {
+		log.Fatal("Error loading .env file")
+	}
+
+	// get env variable
+	DB_USER := os.Getenv("DB_USER")
+	DB_USER_PASSWORD := os.Getenv("DB_USER_PASSWORD")
+	DATABASE := os.Getenv("DATABASE")
+
+	URL := fmt.Sprintf("postgres://%v:%v@localhost/%v?sslmode=disable", DB_USER, DB_USER_PASSWORD, DATABASE)
```
Comment (Jaimin-Parmar) [inline Task5/database/connection.go]: take host from env

**Review by Jaimin-Parmar — CHANGES_REQUESTED** [review]
change as per comment

**Pull request closed without merging**